src/solver.jl

```diff
@@ -4,9 +4,39 @@
 
 using IPFitting, ACE1, LinearAlgebra, JuLIP
 
-export solver_params, generate_solver  
-
-"""TODO documentation"""
+export solver_params   
+
+
+"""
+`solver_params(; kwargs...)` : returns a dictionary containing the
+complete set of parameters required to construct one of the solvers.
+All parameters are passed as keyword argument and the kind of 
+parameters required depend on "type".
+
+### LSQR Parameters 
+* `type = "lsqr"`
```

Comment:
Should this really be `type`?  The examle above seems to indicate that the name of the field that contains the solver type is `solver`

Reply:
agree with Noam here.

Reply:
ok! The reason it was `solver` here in the first place is because that's what `IPFitting` was expecting, but I can change from `type` to `solver` when generating the solver. Updated in next commit.

Reply:
@cortner should it also be changed in IPFitting, then?

Reply:
Elena - fair point. Don’t change it. I’ll give it some thought.